lib/browser/api/menu.js

```diff
@@ -158,23 +158,26 @@ Menu.buildFromTemplate = function (template) {
   if (!Array.isArray(template)) {
     throw new TypeError('Invalid template for Menu')
   }
-
   const menu = new Menu()
+  if (!isValidTemplate(template)) {
+    throw new TypeError('Invalid template for MenuItem: must have at least one of label, role or type')
+  }
   const filtered = removeExtraSeparators(template)
   const sorted = sortTemplate(filtered)
 
-  sorted.forEach((item) => {
-    if (typeof item !== 'object') {
-      throw new TypeError('Invalid template for MenuItem')
-    }
-    menu.append(new MenuItem(item))
-  })
+  sorted.forEach((item) => menu.append(new MenuItem(item)))
 
   return menu
 }
 
 /* Helper Functions */
 
+// validate the template against having the wrong attribute
+function isValidTemplate (template) {
+  return template.every(item =>
+  item != null && typeof item === 'object' && (item.label || item.role || item.type === 'separator'))
```

Comment:
`item.label || item.role ||` doesn't achieve what you're trying to do.
E.g. empty strings and zeros are probably valid labels.
If you want to check presence of a property, please use [`hasOwnProperty()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty) method.